rover_gui/src/QVideoPlayer/QVideoRecorderWidget.cpp

```diff
@@ -0,0 +1,272 @@
+            this,
+            &QVideoRecorderWidget::onScreenshotHandledSuccessfully);
+
+    connect(_startRecordingButton, &QPushButton::clicked, this, &QVideoRecorderWidget::handleRecording);
+    connect(_playerWorkerThreadRecording.get(),
+            &QRecordingWorker::startRecordingHandledSuccessfully,
+            this,
+            &QVideoRecorderWidget::onStartRecordingHandledSuccessfully);
+
+    connect(_playerWorkerThreadRecording.get(),
+            &QRecordingWorker::stopRecordingHandledSuccessfully,
+            this,
+            &QVideoRecorderWidget::onStopRecordingHandledSuccessfully);
+
+    std::optional<std::string> optionalSessionFolderPath = QSessionFolderManager::getInstance().getSessionFolderPath();
+    if (optionalSessionFolderPath.has_value())
+    {
+        _sessionFolderPath = *optionalSessionFolderPath;
+        if (_sessionFolderPath.empty())
+        {
+            QHelper::QToastNotification::getInstance().notifyFromAnyThread("No session folder found",
+                                                                           "SessionFolderManager returned an empty path",
+                                                                           QHelper::QToastNotification::eNotifType::ERROR);
+        }
+    }
+    else
+    {
+        QHelper::QToastNotification::getInstance().notifyFromAnyThread("No session folder found",
+                                                                       "SessionFolderManager couldn't return a valid path",
+                                                                       QHelper::QToastNotification::eNotifType::ERROR);
+    }
+}
+
+QVideoRecorderWidget::~QVideoRecorderWidget() {}
+
+void QVideoRecorderWidget::updateCamURL(std::string& url_)
+{
+    _camURL = url_;
+}
+
+void QVideoRecorderWidget::setCameraControlClientManager(std::shared_ptr<rclcpp::Client<rover_msgs::srv::CameraControl>> client_)
+{
+    if (client_)
+    {
+        _client_cameraControlManager = client_;
+    }
+    else
+    {
+        RCLCPP_WARN(rclcpp::get_logger("GUI"), "Error, couldn't access aruco detection manager client");
+        _screenshotButton->setProperty("class", "error");
+        _screenshotButton->style()->unpolish(_screenshotButton);
+        _screenshotButton->style()->polish(_screenshotButton);
+    }
+}
+
+void QVideoRecorderWidget::handleScreenshot(void)
+{
+    if (_playerWorkerThreadRecording.get() != nullptr)
+    {
+        _playerWorkerThreadRecording->takeScreenshotManager(_client_cameraControlManager,
+                                                            _camURL,
+                                                            _playerIndex,
+                                                            _sessionFolderPath);
+    }
+    else
+    {
+        RCLCPP_ERROR(rclcpp::get_logger("GUI"), "Error, couldn't access Video Player worker");
+    }
+    return;
+}
+
+void QVideoRecorderWidget::handleRecording(void)
```

Comment:
func should be const

```suggestion
void QVideoRecorderWidget::handleRecording(void) const
```